docs/15-lexical-blocking.markdown

```diff
@@ -0,0 +1,381 @@
+facilities, the "block structure" of such languages as Algol, PL/I, 
+SAIL, etc., can be simulated, should you wish to do so.
+
+## 15.1. Basic Considerations
+
+Since what follows appears to be rather complex, a short discussion of 
+the basic problem lexical blocking solves and MDL's basic solution 
+will be given first.
+
+`ATOM`s are identifiers. It is thus essential that whenever you type 
+an `ATOM`, `READ` should respond with the unique identifier you wish 
+to designate. The problem is that it is unreasonable to expect the 
+`PNAME`s of all `ATOM`s to be unique. When you use an `ATOM` `A` in a 
+program, do you mean the `A` you typed two minutes ago, the `A` you 
+used in another one of your programs, or the `A` used by some library 
+program?
+
+Dynamic blocking (pushing down of `LVAL`s) solves many such problems. 
+However, there are some which it does not solve -- such as state 
+variables (whether they are impure or pure). Major problems with a 
+system having only dynamic blocking usually arise only when attempts 
+are made to share large numbers of significant programs among many 
+people.
+
+The solution used in MDL is basically as follows: `READ` must maintain 
+at least one table of `ATOM`s to guarantee any uniqueness. So, MDL 
+allows many such tables and makes it easy for the user to specify 
+which one is wanted. Such a table is an object of `TYPE` `OBLIST` 
+("object list"). All the complication which follows arises out of a 
+desire to provide a powerful, easily used method of working with 
+`OBLIST`s, with reasonable values used by default.
+
+## 15.2. OBLISTs
+
+An `OBLIST` is of `PRIMTYPE` `UVECTOR` with `UTYPE` `LIST`; the `LIST` 
+holds `ATOM`s. The `ATOM`s are ordered by a hash coding on their 
+`PNAME`s: each `LIST` is a hashing bucket.) What follows is 
+information about `OBLIST`s as such.
+
+### 15.2.1. OBLIST Names
+
+Every normally constituted `OBLIST` has a name. The name of an 
+`OBLIST` is an `ATOM` associated with the `OBLIST` under the indicator 
+`OBLIST`. Thus,
+
+    <GETPROP oblist OBLIST>
+
+or
+
+    <GET oblist OBLIST>
+
+returns the name of *oblist*.
+
+Similarly, every name of an `OBLIST` is associated with its `OBLIST`, 
+again under the indicator `OBLIST`, so that
+
+    <GETPROP oblist-name:atom OBLIST>
+
+or
+
+    <GET oblist-name:atom OBLIST>
+
+returns the `OBLIST` whose name is *oblist-name*.
+
+Since there is nothing special about the association of `OBLIST`s and 
+their names, the name of an `OBLIST` can be changed by the use of 
+`PUTPROP`, both on the `OBLIST` and its  name. It is not wise to 
+change the `OBLIST` association without change the name association, 
+since you are likely to confused `READ` and `PRINT` terribly.
+
+You can also use `PUT` or `PUTPROP` to remove the association between 
+an `OBLIST` and its name completely. If you want the `OBLIST` to go 
+away (be garbage collect3ed), *and* you want to keep its name around, 
```

Comment:
typo in "collected"

"and" is underlined in the original, so should be bold (double asterisks) here